setup/openstack/single-instance.yml

```diff
@@ -0,0 +1,91 @@
+heat_template_version: 2018-08-31
+
+description: OpenStack Heat template to deploy a single VM running CapRover.
+
+parameters:
+  image_id:
+    type: string
+    label: Image ID
+    description: Image to be used for compute instance. Ubuntu is recommended.
+  instance_type:
+    type: string
+    label: Instance Type
+    description: Type of instance (flavor) to be used
+  network:
+    type: string
+    label: Network
+    description: Network for the instance
+
+resources:
+  inbound_ssh_security_group:
+    type: OS::Neutron::SecurityGroup
+    properties:
+      name: inbound_ssh_security_group
+      description: Security Group for allowing SSH access
+      rules:
+        - protocol: tcp
+          port_range_min: 22
+          port_range_max: 22
+          remote_ip_prefix: 0.0.0.0/0  # Allows ingress from any IP address
+  inbound_http_security_group:
+    type: OS::Neutron::SecurityGroup
+    properties:
+      name: inbound_http_security_group
+      description: Security Group for allowing HTTP access
+      rules:
+        - protocol: tcp
+          port_range_min: 80
+          port_range_max: 80
+          remote_ip_prefix: 0.0.0.0/0  # Allows ingress from any IP address
+  inbound_https_security_group:
+    type: OS::Neutron::SecurityGroup
+    properties:
+      name: inbound_https_security_group
+      description: Security Group for allowing HTTPS access
+      rules:
+        - protocol: tcp
+          port_range_min: 443
+          port_range_max: 443
+          remote_ip_prefix: 0.0.0.0/0  # Allows ingress from any IP address
+  inbound_caprover_install_security_group:
+    type: OS::Neutron::SecurityGroup
+    properties:
+      name: inbound_caprover_install_security_group
+      description: Security Group for allowing access to port 3000, which is used for CapRover setup.
+      rules:
+        - protocol: tcp
+          port_range_min: 3000
+          port_range_max: 3000
+          remote_ip_prefix: 0.0.0.0/0  # Allows ingress from any IP address
+  caprover_manager:
+    type: OS::Nova::Server
+    properties:
+      key_name: caprover
+      image: { get_param: image_id }
+      flavor: { get_param: instance_type }
+      networks:
+        - network: { get_param: network }
+      security_groups:
+        - default
+        - get_resource: inbound_ssh_security_group
+        - get_resource: inbound_http_security_group
+        - get_resource: inbound_https_security_group
+        - get_resource: inbound_caprover_install_security_group
+      user_data: |
+        #!/bin/sh
+        echo "Installing Docker with commands copied from https://docs.docker.com/engine/install/ubuntu/"
+        # Add Docker's official GPG key:
+        apt-get update
+        apt-get install ca-certificates curl
```

Comment:
I'm not familiar with Openstack, but in a typical setup, this requires user's input (Y to continue). Does openstack automagically bypass this?

Reply:
Good catch! Updated with `-y`. This slipped past my radar because the packages were small enough to not prompt for approval.